proposals/integration-testing.md

```diff
@@ -0,0 +1,110 @@
+# Integration testing tools for BigchainDB
+
+## Problem Description
+We need a proper way to describe, code, and run integration tests. While we have unit tests to verify the correct behavior of functions and methods, we need a way to easily test a network of BigchainDB nodes.
+
+As an example, we want to make sure that if a valid transaction is pushed to the network, after some time that transaction is stored by at least 2/3 of the nodes; an invalid transaction (malformed, double spent, etc.) pushed to the network must be rejected. This must be true also in different scenarios such as not all nodes are up, or there is latency between the nodes. Note that some of those problems has already been addressed by the Tendermint team.
+
+### Use cases
+- Verify we don't break the contract with the user.
+- Verify the behavior of the system in case of failure of its services (MongoDB fails, Tendermint fails, BigchainDB fails)
+- Verify that some properties of the system are implemented correctly.
+
+## Feature: add a client to manage a BigchainDB network
+We define now the characteristics of a new module: the Network Driver, `ND` henceforth. As any other project, it needs a catchy name, but I don't have one in my head now.
+
+The goal of `ND` is to give a simple framework to implement test scenarios that require:
+- creating a network
+- turn the nodes in the network up or down
+- turn the services of a node up or down
+
+There are other useful features we can take into consideration, like:
+- creating and pushing transactions to one or more nodes in the network
+- check the existence of a transaction in one or more nodes in the network
+
+*Transaction specific* features can be added to `ND` later. We can just use the BigchainDB Python driver for now, or raw `HTTP` requests.
+
+[Bitcoin functional tests](https://github.com/bitcoin/bitcoin/tree/v0.15.0/test/functional) have as similar approach (thanks @codegeschrei for the link).
+
+## Proposed Change
+First of all, a new directory will be created, namely `integration-tests`, in the root of the project. This directory will contain the code for `ND`, and will be home for the new integration tests.
+
+Integration tests require not only one node, but a whole network to be up and running. A `docker-compose` configuration file has been created for that purpose, and the `ND` module should leverage it.
+
+The `ND` module implements two main functions:
+- network setup, using `docker-compose`
+- basic transaction management
+
+In the next sections we will use `ND` to refer to the Python class, and `network` to refer to an instance of it. For now, `network` will be likely to be a singleton, since it will control `docker` in the current host. This **will not** be the final name of the module.
```

Comment:
> will be likely to be a singleton

I wonder if this could be reworded a bit into something like:

> will most likely be a singleton

or some other variant ...